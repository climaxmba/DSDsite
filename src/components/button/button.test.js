import { render, screen, fireEvent } from '@testing-library/react';
import Button from './Button';

jest.mock('./button.module.css', () => ({
  button: 'button',
  buttonPrimary: 'button--primary',
  buttonIcon: 'button__icon',
}));

test('renders correct text', () => {
  render(<Button buttonText='Click me' />);
  const buttonElement = screen.getByText(/click me/i);
  expect(buttonElement).toBeInTheDocument();
});

test('calls onClick prop when clicked', () => {
  const handleClick = jest.fn();
  render(<Button buttonText='Click me' onClick={handleClick} />);
  fireEvent.click(screen.getByText(/click me/i));
  expect(handleClick).toHaveBeenCalledTimes(1);
});

test('renders correctly with primary variant', () => {
  const { asFragment } = render(
    <Button buttonText='Primary Button' variant='primary' />
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders correctly with non-primary variant', () => {
  const { asFragment } = render(
    <Button buttonText='Secondary Button' variant='secondary' />
  );
  expect(asFragment()).toMatchSnapshot();
});

test('shows icon when showIcon is true', () => {
  render(<Button buttonText='Button with Icon' showIcon />);
  const iconElement = screen.getByTestId('button-icon');
  expect(iconElement).toBeInTheDocument();
});
