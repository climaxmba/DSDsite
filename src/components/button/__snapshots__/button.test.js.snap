// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with non-primary variant 1`] = `
<DocumentFragment>
  <button
    class="button "
    data-testid="button-secondary"
  >
    Secondary Button
  </button>
</DocumentFragment>
`;

exports[`renders correctly with primary variant 1`] = `
<DocumentFragment>
  <button
    class="button undefined"
    data-testid="button-primary"
  >
    Primary Button
  </button>
</DocumentFragment>
`;
